freqtrade/freqtradebot.py

```diff
@@ -255,6 +278,16 @@ def _refresh_active_whitelist(self, trades: List[Trade] = []) -> List[str]:
         self.pairlists.refresh_pairlist()
         _whitelist = self.pairlists.whitelist
 
+        # If external signal leader, broadcast whitelist data
+        # Should we broadcast before trade pairs are added?
+
+        if self.external_signal_controller:
+            if self.external_signal_controller.is_leader():
+                self.rpc.emit_data({
+                    "data_type": LeaderMessageType.pairlist,
+                    "data": _whitelist
+                })
+
```

Comment:
i think this check should be moved to the rpchandler.
the bot just issues a "emit_data" event - it's then up to the handler to decide if an external controller is available or not.
That'll align this pattern to how we're doing it with send_msg.

Reply:
yes - that's the idea
in theory, you could have some random other RPC module that's loaded and does "something" with this messagetype (could be sending a telegram message "new whitelist emitted").

It's not up to the bot to determine if that really should be sent or not (please take the above as a theoretical example ...)